### PR TITLE
update graph docs up to Reusable Components

### PR DESCRIPTION
**Pull request by rkuhn**
actually covers all of stream-graphs.rst now

### CODE REVIEW

File: akka-docs-dev/rst/scala/code/docs/stream/FlowGraphDocSpec.scala [hunk 1]
```diff
@@ -20,13 +20,13 @@ class FlowGraphDocSpec extends AkkaSpec {
   "build simple graph" in {
     //format: OFF
     //#simple-flow-graph
-    val g = FlowGraph.closed() { implicit b =>
+    val g = FlowGraph.closed() { implicit builder: FlowGraph.Builder =>
```
Comment (drewhk) [inline akka-docs-dev/rst/scala/code/docs/stream/FlowGraphDocSpec.scala]: good idea to show the type in the first samples

File: akka-docs-dev/rst/scala/code/docs/stream/FlowGraphDocSpec.scala [hunk 2]
```diff
@@ -207,12 +207,13 @@ class FlowGraphDocSpec extends AkkaSpec {
     import FanInShape.Name
     import FanInShape.Init
 
-    case class PriorityWorkerPoolShape2[In, Out](
-      _init: Init[Out] = Name("PriorityWorkerPool")) extends FanInShape2[In, In, Out](_init) {
+    class PriorityWorkerPoolShape2[In, Out](_init: Init[Out] = Name("PriorityWorkerPool"))
```
Comment (drewhk) [inline akka-docs-dev/rst/scala/code/docs/stream/FlowGraphDocSpec.scala]: thanks for the correction

File: akka-docs-dev/rst/scala/stream-graphs.rst [hunk 3]
```diff
@@ -27,19 +28,19 @@ Akka Streams currently provide these junctions:
 
 * **Fan-out**
 
- - ``Broadcast[T]`` – (1 input, n outputs) signals each output given an input signal,
- - ``Balance[T]`` – (1 input => n outputs), signals one of its output ports given an input signal,
- - ``UnZip[A,B]`` – (1 input => 2 outputs), which is a specialized element which is able to split a stream of ``(A,B)`` tuples into two streams one type ``A`` and one of type ``B``,
- - ``FlexiRoute[In]`` – (1 input, n outputs), which enables writing custom fan out elements using a simple DSL,
+ - ``Broadcast[T]`` – *(1 input, N outputs)* signals each output given an input signal
```
Comment (drewhk) [inline akka-docs-dev/rst/scala/stream-graphs.rst]: hm. signal -> element? signals -> emits?

File: akka-docs-dev/rst/scala/stream-graphs.rst [hunk 4]
```diff
@@ -27,19 +28,19 @@ Akka Streams currently provide these junctions:
...
- - ``ZipWith[A,B,...,Out]`` – (n inputs (defined upfront), 1 output), which takes a function of n inputs that, given all inputs are signalled, transforms and emits 1 output,
- - ``Zip[A,B]`` – (2 inputs, 1 output), which is a :class:`ZipWith` specialised to zipping input streams of ``A`` and ``B`` into an ``(A,B)`` tuple stream,
- - ``Concat[A]`` – (2 inputs, 1 output), which enables to concatenate streams (first consume one, then the second one), thus the order of which stream is ``first`` and which ``second`` matters,
- - ``FlexiMerge[Out]`` – (n inputs, 1 output), which enables writing custom fan-in elements using a simple DSL.
+ - ``Merge[In]`` – *(N inputs , 1 output)* picks randomly from inputs pushing them one by one to its output
+ - ``MergePreferred[In]`` – like :class:`Merge` but if elements are available on ``preferred`` port, it picks from it, otherwise randomly from ``others``
+ - ``ZipWith[A,B,...,Out]`` – *(N inputs, 1 output)* which takes a function of N inputs that given a value for each input emits 1 output element
```
Comment (drewhk) [inline akka-docs-dev/rst/scala/stream-graphs.rst]: strictly speaking ZipWithN

File: akka-docs-dev/rst/scala/stream-graphs.rst [hunk 5]
```diff
@@ -58,20 +59,29 @@ will be inferred.
    Junction *reference equality* defines *graph node equality* (i.e. the same merge *instance* used in a FlowGraph
    refers to the same location in the resulting graph).
 
-Notice the ``import FlowGraphImplicits._`` which brings into scope the ``~>`` operator (read as "edge", "via" or "to").
+Notice the ``import FlowGraph.Implicits._`` which brings into scope the ``~>`` operator (read as "edge", "via" or "to").
```
Comment (drewhk) [inline akka-docs-dev/rst/scala/stream-graphs.rst]: mention the reverse arrow, too

File: akka-docs-dev/rst/scala/stream-graphs.rst [hunk 6]
```diff
@@ -82,53 +92,59 @@ Constructing and combining Partial Flow Graphs
...
 complex element and from there on treat it as simple compound stage for linear computations.
 
 In order to create a Source from a partial flow graph ``Source`` provides a special apply method that takes a function
-that must return an ``UndefinedSink``. This undefined sink will become "the sink that must be attached before this Source
-can run". Refer to the example below, in which we create a Source that zips together two numbers, to see this graph
+that must return an :class:`Outlet[T]`. This unconnected sink will become “the sink that must be attached before this Source
+can run”. Refer to the example below, in which we create a Source that zips together two numbers, to see this graph
 construction in action:
 
 .. includecode:: code/docs/stream/StreamPartialFlowGraphDocSpec.scala#source-from-partial-flow-graph
```
Comment (drewhk) [inline akka-docs-dev/rst/scala/stream-graphs.rst]: these samples use explicit Flows as edges. Do we want to show the shorthand version where you chain from ports?

**Comment by drewhk** [comment]
LGTM

File: akka-docs-dev/rst/scala/stream-graphs.rst (hunk 6 above)
Comment (rkuhn) [inline akka-docs-dev/rst/scala/stream-graphs.rst]: already changed

File: akka-docs-dev/rst/scala/stream-graphs.rst (hunk 4 above)
Comment (rkuhn) [inline akka-docs-dev/rst/scala/stream-graphs.rst]: there is no such thing anymore ;-)

**Pull request merged**